Samples/Samples/ViewModel/CalendarEventAddViewModel.cs

```diff
@@ -152,6 +236,267 @@ public TimeSpan EndTime
+
+        public bool IsYearly => SelectedRecurrenceType == RecurrenceFrequency.Yearly.ToString() || SelectedRecurrenceType == RecurrenceFrequency.YearlyOnDay.ToString();
+
+        bool isEveryWeekday;
+
+        public bool IsEveryWeekday
+        {
+            get => isEveryWeekday;
+            set
+            {
+                if (SetProperty(ref isEveryWeekday, value))
+                {
+                    if (value)
+                    {
+                        RecurrenceInterval = 0;
+                        RecurrenceDays = recurrenceWeekdays;
+                    }
+                    else
+                    {
+                        RecurrenceInterval = 1;
+                    }
+                }
+            }
+        }
+
+        bool isMonthDaySpecific = true;
+
+        public bool IsMonthDaySpecific
+        {
+            get => isMonthDaySpecific;
+            set
+            {
+                if (SetProperty(ref isMonthDaySpecific, value))
+                {
+                    if (value)
+                    {
+                        SelectedRecurrenceMonthDay = 1;
+                        SelectedRecurrenceMonthWeek = string.Empty;
+                        SelectedMonthWeekRecurrenceDay = string.Empty;
+                    }
+                    else
+                    {
+                        SelectedRecurrenceMonthDay = 0;
+                        SelectedRecurrenceMonthWeek = "First";
+                        SelectedMonthWeekRecurrenceDay = DayOfTheWeek.Monday.ToString();
+                    }
+                }
+            }
+        }
+
+        public ObservableCollection<DayOfTheWeekSwitch> RecurrenceDays { get; set; } = new ObservableCollection<DayOfTheWeekSwitch>()
+        {
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Monday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Tuesday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Wednesday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Thursday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Friday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Saturday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Sunday }
+        };
+
+        public ObservableCollection<string> RecurrenceTypes { get; } = new ObservableCollection<string>()
+        {
+            RecurrenceFrequency.None.ToString(),
+            RecurrenceFrequency.Daily.ToString(),
+            RecurrenceFrequency.Weekly.ToString(),
+            RecurrenceFrequency.Monthly.ToString(),
+            RecurrenceFrequency.Yearly.ToString()
+        };
+
+        string selectedRecurrenceType = RecurrenceFrequency.None.ToString();
```

Comment:
Wouldn't it be better to have this as a RecurrencyFrequency rather than a string?

Reply:
Adjusted in latest commit, as well as fixed all other related parameters that were using strings instead of well formed enums.